dsl/pace/dsl/dace/orchestrate.py

```diff
@@ -0,0 +1,305 @@
+            self.daceprog = methodwrapper.__get__(obj_to_bind)
+
+        @property
+        def global_vars(self):
+            return self.daceprog.global_vars
+
+        @global_vars.setter
+        def global_vars(self, value):
+            self.daceprog.global_vars = value
+
+        def __call__(self, *args, **kwargs):
+            if self.lazy_method.use_dace:
+                sdfg = self.__sdfg__(*args, **kwargs)
+                return call_sdfg(
+                    self.daceprog,
+                    sdfg,
+                    args,
+                    kwargs,
+                    sdfg_final=(self.lazy_method._load_sdfg is not None),
+                )
+            else:
+                return self.lazy_method.func(self.obj_to_bind, *args, **kwargs)
+
+        def __sdfg__(self, *args, **kwargs):
+            if self.lazy_method._load_sdfg is None:
+                return self.daceprog.to_sdfg(
+                    *args,
+                    **self.daceprog.__sdfg_closure__(),
+                    **kwargs,
+                    save=False,
+                    simplify=False,
+                )
+            else:
+                if os.path.isfile(self.lazy_method._load_sdfg):
+                    self.daceprog.load_sdfg(
+                        self.lazy_method._load_sdfg, *args, **kwargs
+                    )
+                else:
+                    self.daceprog.load_precompiled_sdfg(
+                        self.lazy_method._load_sdfg, *args, **kwargs
+                    )
+                return self.daceprog.__sdfg__(*args, **kwargs)
+
+        def __sdfg_closure__(self, reevaluate=None):
+            return self.daceprog.__sdfg_closure__(reevaluate)
+
+        def __sdfg_signature__(self):
+            return self.daceprog.argnames, self.daceprog.constant_args
+
+        def closure_resolver(self, constant_args, given_args, parent_closure=None):
+            return self.daceprog.closure_resolver(
+                constant_args, given_args, parent_closure
+            )
+
+    def __init__(self, func, load_sdfg):
+        self.func = func
+        self._load_sdfg = load_sdfg
+
+    def __get__(self, obj, objype=None) -> SDFGEnabledCallable:
+        """Return SDFGEnabledCallable wrapping original obj.method from cache.
+        Update cache first if need be"""
+        if (id(obj), id(self.func)) not in LazyComputepathMethod.bound_callables:
+
+            LazyComputepathMethod.bound_callables[
+                (id(obj), id(self.func))
+            ] = LazyComputepathMethod.SDFGEnabledCallable(self, obj)
+
+        return LazyComputepathMethod.bound_callables[(id(obj), id(self.func))]
+
+    @property
+    def use_dace(self):
+        return dace_config.get_orchestrate()
+
+
+def computepath_method(
+    method: Callable[..., Any], load_sdfg: Optional[str] = None
+) -> Union[Callable[..., Any], LazyComputepathFunction]:
+    """Decorator wrapping a class method in a JIT DaCe orchestrator"""
```

Comment:
optional: Handy since these are the entrypoint functions we'll be using.

```suggestion
    """
    Decorator wrapping a class method in a JIT DaCe orchestrator.
    
    Args:
        method: class method to either orchestrate or directly execute
        load_sdfg: folder path to a pre-compiled SDFG or file path to a .sdfg graph
            that will be compiled but not regenerated.
"""
```